pterodactyl/client/servers/files.md

```diff
@@ -0,0 +1,259 @@
+
+| Code | Description                    |
+| ---- | ------------------------------ |
+| 200  | List of files in the directory |
+
+### Example Response
+
+```json
+{
+  "data": [
+    {
+      "object": "file_object",
+      "attributes": {
+        "name": "example_file.txt",
+        "mode": "755",
+        "mode_bits": "rwxr-xr-x",
+        "size": 1024,
+        "is_file": true,
+        "is_symlink": false,
+        "mimetype": "text/plain",
+        "created_at": "2023-06-10T08:30:00Z",
+        "modified_at": "2023-06-10T10:15:00Z"
+      }
+    }
+  ]
+}
+```
+
+### `GET /api/client/servers/:serverId/files/contents`
+
+Returns the contents of a specified file for the user.
+
+### Body
+
+| Name | Visibility | Type   | Description      |
+| ---- | ---------- | ------ | ---------------- |
+| file | Required   | string | Path to the file |
+
+### Responses
+
+| Code | Description  |
+| ---- | ------------ |
+| 200  | File content |
+
+### `GET /api/client/servers/:serverId/files/download`
+
+Generates a one-time token with a link that the user can use to download a given file.
+
+### Body
+
+| Name | Visibility | Type   | Description      |
+| ---- | ---------- | ------ | ---------------- |
+| file | Required   | string | Path to the file |
+
+### Responses
+
+| Code | Description                  |
+| ---- | ---------------------------- |
+| 200  | Signed URL for file download |
+
+### Example Response
+
+```json
+{
+  "object": "signed_url",
+  "attributes": {
+    "url": "https://pterodactyl.file.properties:3000/download/file?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJmaWxlX3BhdGgiOiJkZXNrdG9wL3Rlc3QucGRmIiwic2VydmVyX3V1aWQiOiJ1c2VyLWlkIn0.u6Ln6vZDP1Z8fxSoZ7i7J9RTx69R5_Hn6mIF5xGm4O8"
+  }
+}
+```
+
+### `PUT /api/client/servers/:serverId/files/rename`
+
+Renames a file on the remote machine.
+
+### Body
+
+| Name    | Visibility | Type          | Description                |
+| ------- | ---------- | ------------- | -------------------------- |
+| root    | Required   | string        | Root directory             |
+| files[] | Required   | array[string] | Files to rename (multiple) |
+
+### Responses
+
+| Code | Description  |
+| ---- | ------------ |
+| 204  | File renamed |
+
+### `POST /api/client/servers/:serverId/files/copy`
+
+Copies a file on the server.
+
+### Body
+
+| Name     | Visibility | Type   | Description          |
+| -------- | ---------- | ------ | -------------------- |
+| location | Required   | string | Location of the file |
+
+### Responses
+
+| Code | Description |
+| ---- | ----------- |
+| 204  | File copied |
+
+### `POST /api/client/servers/:serverId/files/write`
+
+Writes the contents of the specified file to the server.
+
+### Body
+
+| Name | Visibility | Type   | Description                       |
+| ---- | ---------- | ------ | --------------------------------- |
+| file | Required   | string | File to write                     |
+| -    | Required   | string | Content of the file to be written |
```

Comment:
We can make an exception for this endpoint – the request body _is_ the content that is being written, there is no particular format that is required for this but most API libraries/wrappers use `text/plain`. Perhaps a message indicating this would be better.

Reply:
I have addressed the requested changes.